### PR TITLE
Updated Pact guide

### LINKED ISSUE #11430: Document Pact integration workflow and failure resolutions
## Workflow

Document rules like, "Don't merge Pact tests until they have been successfully verified and the `can-i-deploy` job passes."

## Failure resolutions

Confirm if these scenarios occur in CI after setting up the webhooks and jobs.

For failures, come up with a standard resolution and document.

1. `vets-website` PR passes

2. `vets-website` PR passes
   `vets-api` introduces breaking change for PR (but not breaking for master)
    both fail if PR merges (maybe only `vets-website` fails due to pending pacts feature?).

    Presumably, this can be reproduced by implementing a change in `vets-api` that accommodates a new contract from the `vets-website` PR and then rolling it back.

3. `vets-website` PR fails initially
   `vets-api` implements feature
   `vets-website` PR passes

4. `vets-api` PR passes verification

5. `vets-api` PR fails verification (introduces breaking change)

### PR DESCRIPTION
**Pull request by U-DON**
Fixes #11430.

### Changes
- Re-wrote sections on workflow and naming conventions.
- Consolidated some content into other sections.
- Fixed typos.

### CODE REVIEW

File: platform/testing/contract-testing/how-to-use-pact-for-contract-testing.md [hunk 1]
```diff
@@ -54,21 +50,22 @@ Pact is a tool that enables consumer driven contract testing (CDCT) by defining
...
-* **Consumer** -- The consumer `vets-website` generates pacts.
-* **Provider** -- The provider `vets-api` verifies pacts.
-* **Pact** -- A contract between a consumer and provider is called a _pact_. Each pact is a collection of _interactions_.
-* **Interaction** -- A request and response pair.
-* **Broker** -- Central location where pacts are hosted. The Pact broker is currently hosted on [Heroku](https://vagov-pact-broker.herokuapp.com/). You can view the interactions per endpoint and the verification matrix from the broker index.
+- **consumer**: The consumer (`vets-website`) in the contract makes requests. In the Pact testing process, it generates pacts.
+- **provider**: The provider (`vets-api`) in the contract provides responses. In the Pact testing process, it verifies pacts.
+- **pact**: A contract between a consumer and provider is called a _pact_. Each pact is a collection of _interactions_.
+- **interaction**: A request and response pair.
+- **provider state**: The description of a state that the consumer expects the provider to be in. In a test, the provider handles a state by doing the necessary setup to acccommodate a request.
```
Comment (LindseySaari) [inline platform/testing/contract-testing/how-to-use-pact-for-contract-testing.md:63]: Nice! I like this addition

File: platform/testing/contract-testing/how-to-use-pact-for-contract-testing.md [hunk 2]
```diff
@@ -608,9 +621,9 @@ https://vagov-pact-broker.herokuapp.com/pacts/provider/VA.gov%20API/consumer/HCA
 
 After the vets-api feature branch is merged to master, that pact should be able to pass verification.
 
-The CircleCI build for vets-api is still under construction with respect to parallelizing our test suite. Tests currently fail simplecov minimum coverage standards due to processes overriding results from other parallel test processes. A solution is in the works to properly merge simplecov results. 
+The CircleCI build for vets-api is still under construction with respect to parallelizing our test suite. Tests currently fail simplecov minimum coverage standards due to processes overriding results from other parallel test processes. A solution is in the works to properly merge simplecov results.
 
-The pact verification functionality is in working order for developer usage when a commit is pushed or when invoked via webhook from the pact broker during a contract content changed event. 
+The pact verification functionality is in working order for developer usage when a commit is pushed or when invoked via webhook from the pact broker during a contract content changed event.
```
Comment (LindseySaari) [inline platform/testing/contract-testing/how-to-use-pact-for-contract-testing.md:626]: maybe add something here like: The webhook/pact verification results can be viewed in Circle.

Comment (LindseySaari) [reply platform/testing/contract-testing/how-to-use-pact-for-contract-testing.md:626]: I also added functionality to the pact_verification workflow when triggered via webhook to create an artifact containing consumer tags and other information

File: platform/testing/contract-testing/how-to-use-pact-for-contract-testing.md [hunk 3]
```diff
@@ -77,76 +74,84 @@ VFS teams are required to use Pact for contract testing in the following scenari
...
-![](https://i.imgur.com/zQMyDS0.png)
+1. BE engineers update API endpoints and/or provider states in a feature branch in vets-api.
+2. The vets-api CI pipeline runs the verification job, which verifies the all of the latest pacts on the vets-website master branch.
+3. If the verification job passes, the branch can be merged.
+4. If the verification has failed for any pact, BE engineers should discuss with FE engineers to resolve the failure.
+   BE should adjust provider states or API responses to accommodate the pact or FE should update the pact.
+   If the resolution only involves FE changes, wait for those changes to get merged and re-run the vets-api CI pipeline.
 
-### Implementation details
+![](https://i.imgur.com/zQMyDS0.png)
```
Comment (LindseySaari) [inline platform/testing/contract-testing/how-to-use-pact-for-contract-testing.md:112]: Do we like having this large image here? I'm wondering if it's helpful

**Review by LindseySaari — COMMENTED** [review]
Wondering if we should include information on the tentative plans to move the broker into the AWS infra and how this will change the pact verification and circle

File: platform/testing/contract-testing/how-to-use-pact-for-contract-testing.md [hunk 4]
```diff
@@ -77,76 +74,84 @@ VFS teams are required to use Pact for contract testing in the following scenari
...
+6. If verification has failed for any pact, FE engineers should discuss with BE engineers to resolve the failure.
+   **The branch should not be merged until the `can-i-deploy` task is successful.**
+   FE should adjust the pact or BE should update provider states or API responses to accommodate the pact.
+   If the resolution only involves BE changes, wait for those changes to get merged and re-run the vets-website CI pipeline.
 
+#### Back-end workflow
 
-![](https://i.imgur.com/zQMyDS0.png)
+1. BE engineers update API endpoints and/or provider states in a feature branch in vets-api.
+2. The vets-api CI pipeline runs the verification job, which verifies the all of the latest pacts on the vets-website master branch.
```
Comment (LindseySaari) [inline platform/testing/contract-testing/how-to-use-pact-for-contract-testing.md:106]: typo here: "the all of the"

Comment (LindseySaari) [reply platform/testing/contract-testing/how-to-use-pact-for-contract-testing.md:106]: Maybe add something about how you can view the pact verification results in CircleCI

File: platform/testing/contract-testing/how-to-use-pact-for-contract-testing.md [hunk 5]
```diff
@@ -77,76 +74,84 @@ VFS teams are required to use Pact for contract testing in the following scenari
...
+In that pact, there is an interaction with this description:
+> Given **at least one matching result exists**, upon receiving **a search query** from **Search**, with `<request>` **VA.gov API** will respond with `<response>`.
 
-#### Naming Conventions
+Note how the description is structured according to this pattern:
+> Given `<provider state>`, upon receiving `<request description>` from `<consumer>`, with `<request object>` `<provider>` will respond with `<response object>`.
 
-The purpose of contract testing is to ensure that the consumer and provider have a shared understanding of the messages that will pass between them. **To that end, we follow a policy of explicit or descriptive naming when defining interactions and provider states.** The goal is for developers to understand any interaction in a pact. 
+When defining pacts and their interactions, it's important to consider how you name these key attributess: **provider state, request description, consumer, and provider**.
```
Comment (LindseySaari) [inline platform/testing/contract-testing/how-to-use-pact-for-contract-testing.md:126]: typo: attributess

**Comment by LindseySaari** [comment]
A few comments inline, but these updates look great. I really like the re-organization and simplification!

File: platform/testing/contract-testing/how-to-use-pact-for-contract-testing.md (hunk 2 above)
Comment (U-DON) [reply platform/testing/contract-testing/how-to-use-pact-for-contract-testing.md]: Nice improvement! I've updated this section on both counts.

File: platform/testing/contract-testing/how-to-use-pact-for-contract-testing.md (hunk 3 above)
Comment (U-DON) [reply platform/testing/contract-testing/how-to-use-pact-for-contract-testing.md:112]: I suppose we could do better, but the diagram I made a while back is a bit outdated now. This generic diagram is what we've got for now, so I'll leave it alone.

File: platform/testing/contract-testing/how-to-use-pact-for-contract-testing.md (hunk 4 above)
Comment (U-DON) [reply platform/testing/contract-testing/how-to-use-pact-for-contract-testing.md]: I intended for this section to be more conceptual, but I added information about viewing the verification results in CircleCI in another section.

**Comment by U-DON** [comment]
Probably shouldn't add any speculation or discussion points on the Pact broker is going to live. We can update the doc when any changes become official.

**Pull request merged**